### PR TITLE
Add pocket

### PR DESCRIPTION
**Pull request by gastonabella**
### Brief Description
Add pocket service for frontend.
### Related card or ticket
https://github.com/Pis-moove-it/reciclando-backend/issues/10
### Additional Details
Given a route_id, a collection_point_id and an array of pocket_serial_numbers, it creates a Collection associated to the ids mentioned, and create pockets with the serial numbers given associated to the collection too.
- Service for frontend
- Controller tests
- Swagger service
- Fixtures done for routes and collection points

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@
   end
 
   resources :pockets, only: [:index]
+  
+  resources :routes, only: [] do
+    resources :collections, only: %i[create]
+  end
+
+  resources :routes, only: [] do
```
Comment (fbalarini) [inline config/routes.rb]: Duplicated

File: db/seeds.rb [hunk 2]
```diff
@@ -4,6 +4,10 @@
 require_relative 'fixtures/admin_users'
 require_relative 'fixtures/organizations'
 require_relative 'fixtures/users'
+require_relative 'fixtures/routes'
+require_relative 'fixtures/collection_points'
+
+
```
Comment (fbalarini) [inline db/seeds.rb]: remove this whitespace

File: spec/controllers/bales_controller_spec.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@ def create_bale_call(weight, material)
         expect(response).to have_http_status(:bad_request)
       end
       it 'does not create bales without weight' do
-        create_bale_call(bale.weight, nil)
+        create_bale_call(nil, bale.material)
```
Comment (fbalarini) [inline spec/controllers/bales_controller_spec.rb:28]: why?

File: spec/factories/collection_points.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :collection_point do
-    latitude 'MyString'
-    longitude 'MyString'
+    latitude { Faker::Number.decimal(2, 6) }
```
Comment (fbalarini) [inline spec/factories/collection_points.rb]: mmm, Faker doesn't have a method for longitude and latitude ?

File: app/controllers/collections_controller.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class CollectionsController < AuthenticateController
+  def create
+    collection = Collection.new(collection_params.except(:pocket_serial_numbers).merge(route: route))
+    serials = params[:collection][:pocket_serial_numbers]
```
Comment (fbalarini) [inline app/controllers/collections_controller.rb:4]: maybe it is better to get this from a private method in this controller

File: app/controllers/collections_controller.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    pockets = serials.collect { |s| { serial_number: s, organization: logged_user.organization } }
+    collection.pockets.new(pockets)
```
Comment (fbalarini) [inline app/controllers/collections_controller.rb:6]: Research about `nested_attributes`. It is possible to make this easier, if pockets is accepted as a nested attribute for collections, you can create them by putting pockets params inside collection params. If you don't get my point, let me know and we discuss it.

**Review by fbalarini — CHANGES_REQUESTED** [review]

File: spec/controllers/bales_controller_spec.rb (hunk 3 above)
Comment (gastonabella) [reply spec/controllers/bales_controller_spec.rb:28]: This is an error of an old feature and I fix it as soon as I noticed it. Perhaps this should have been done in other PR.

**Pull request closed without merging**